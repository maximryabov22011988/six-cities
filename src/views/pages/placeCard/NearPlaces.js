import React from 'react';
import PropTypes from 'prop-types';

import PlaceCardList from '../../components/PlaceCardList';

const propTypes = {
  offers: PropTypes.arrayOf(
    PropTypes.shape({
      bedrooms: PropTypes.number,
      city: PropTypes.shape({
        name: PropTypes.string,
        location: PropTypes.object,
      }),
      description: PropTypes.string,
      goods: PropTypes.arrayOf(PropTypes.string),
      host: PropTypes.shape({
        avatar_url: PropTypes.string,
        id: PropTypes.number,
        is_pro: PropTypes.bool,
        name: PropTypes.string,
      }),
      id: PropTypes.number,
      images: PropTypes.arrayOf(PropTypes.string),
      is_favorite: PropTypes.bool,
      is_premium: PropTypes.bool,
      location: PropTypes.shape({
        latitude: PropTypes.number,
        longitude: PropTypes.number,
        zoom: PropTypes.number,
      }),
      max_adults: PropTypes.number,
      preview_image: PropTypes.string,
      price: PropTypes.number,
      rating: PropTypes.number,
      title: PropTypes.string,
      type: PropTypes.string,
    }),
  ).isRequired,
};

function NearPlaces({ offers }) {
  return (
    <div className="container">
      <section className="near-places places">
        <h2 className="near-places__title">Other places in the neighbourhood</h2>
        <PlaceCardList offers={offers} />
      </section>
    </div>
  );
}

NearPlaces.propTypes = propTypes;

export default NearPlaces;
