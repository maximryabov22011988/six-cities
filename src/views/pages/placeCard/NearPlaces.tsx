import * as React from 'react';

import PlaceCardList from '../../components/PlaceCardList';

import { Offer, onAddToFavorities, onRemoveFromFavorities } from '../../types';

interface Props {
  offers: Array<Offer>;
  onAddToFavorities: onAddToFavorities;
  onRemoveFromFavorities: onRemoveFromFavorities;
}

function NearPlaces({ offers, onAddToFavorities, onRemoveFromFavorities }: Props) {
  return (
    <div className="container">
      <section className="near-places places">
        <h2 className="near-places__title">Other places in the neighbourhood</h2>
        <PlaceCardList
          offers={offers}
          parentClassName="near-places"
          onAddToFavorities={onAddToFavorities}
          onRemoveFromFavorities={onRemoveFromFavorities}
        />
      </section>
    </div>
  );
}

export default NearPlaces;
