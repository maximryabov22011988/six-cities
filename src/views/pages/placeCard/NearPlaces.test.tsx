import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";

import NearPlaces from './NearPlaces';
import offer from '../../mocks/offer';

const offers = [offer];

it('NearPlaces renders correctly with mandatory props', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <NearPlaces
          offers={offers}
          onAddToFavorities={jest.fn()}
          onRemoveFromFavorities={jest.fn()}
        />
      </BrowserRouter>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
