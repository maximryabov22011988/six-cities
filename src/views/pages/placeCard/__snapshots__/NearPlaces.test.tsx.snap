// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NearPlaces renders correctly with mandatory props 1`] = `
<div
  className="container"
>
  <section
    className="near-places places"
  >
    <h2
      className="near-places__title"
    >
      Other places in the neighbourhood
    </h2>
    <div
      className="places__list near-places__list"
    >
      <article
        className="place-card near-places__card"
      >
        <div
          className="near-places__image-wrapper place-card__image-wrapper"
        >
          <a
            href="#"
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "background": "#e1e1e1",
                  "borderRadius": "4px",
                  "display": "flex",
                  "height": "200px",
                  "justifyContent": "center",
                  "width": "260px",
                }
              }
            >
              <img
                alt="Place image"
                className="place-card__image"
                height={0}
                onError={[Function]}
                onLoad={[Function]}
                src="https://es31-server.appspot.com/six-cities/static/hotel/17.jpg"
                width={0}
              />
              <span>
                Loading ...
              </span>
            </div>
          </a>
        </div>
        <div
          className="place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                €
                161
              </b>
              <span
                className="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              className="place-card__bookmark-button button"
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="place-card__rating rating"
          >
            <div
              className="place-card__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "54.00%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            className="place-card__name"
          >
            <a
              href="/offer/1"
              onClick={[Function]}
            >
              Canal View Prinsengracht
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            house
          </p>
        </div>
      </article>
    </div>
  </section>
</div>
`;
